src/main/java/com/adamk33n3r/runelite/watchdog/TrayNotifier.java

```diff
@@ -0,0 +1,259 @@
+package com.adamk33n3r.runelite.watchdog;
+
+import com.google.common.base.Strings;
+import com.google.common.escape.Escaper;
+import com.google.common.escape.Escapers;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.client.RuneLite;
+import net.runelite.client.RuneLiteProperties;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.ui.ClientUI;
+import net.runelite.client.util.OSType;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.awt.*;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This file is as-much-as-possible a copy of {@link net.runelite.client.Notifier}
+ * The copied functions can be run through a differ on Notifier's versions of the functions
+ */
+@Slf4j
+@Singleton
+public class TrayNotifier {
+    @Inject
+    protected transient ClientUI clientUI;
+
+    @Inject
+    protected transient Client client;
+
+    @Inject
+    protected transient AlertManager alertManager;
+
+    @Inject
+    protected transient WatchdogConfig watchdogConfig;
+
+    @Inject
+    protected transient ConfigManager configManager;
+    private static final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(2);
+    private final transient FakeRuneLiteConfig runeLiteConfig;
+    private transient final String appName = "RuneLite";
+    // Copied from RuneLite's Notifier class
+    private final transient Path notifyIconPath = RuneLite.RUNELITE_DIR.toPath().resolve("icon.png");
+    private transient boolean terminalNotifierAvailable;
+    private static final String DOUBLE_QUOTE = "\"";
+    private static final Escaper SHELL_ESCAPE = Escapers.builder()
+            .addEscape('"', "'")
+            .build();
+
+    public TrayNotifier() {
```

Comment:
Could you not do similar to Notifier where it injects all the dependencies here? Confused what the `FakeRuneLiteConfig` is for. We can't just use the real one?

Reply:
You're right, this works as expected! Fixed in https://github.com/adamk33n3r/runelite-watchdog/pull/112/commits/4c25f9698e9c76f63424e4f7f415c6c35ef4cfa4